contracts/settling_game/modules/relics/Relics.cairo

```diff
@@ -97,36 +130,226 @@ func set_relic_holder{pedersen_ptr: HashBuiltin*, syscall_ptr: felt*, range_chec
         tempvar pedersen_ptr = pedersen_ptr;
     }
 
-    // Capture back if held by loser
-    // Check if loser has winners token
-    let (winners_relic_owner) = get_current_relic_holder(winner_token_id);
+    let (realms_address) = IModuleController.get_external_contract_address(
+        controller, ExternalContractIds.Realms
+    );
 
-    let (holds_relic) = uint256_eq(winners_relic_owner, loser_token_id);
+    let (winner_realm_data: RealmData) = IRealms.fetch_realm_data(realms_address, winner_token_id);
 
-    if (holds_relic == TRUE) {
-        _set_relic_holder(winner_token_id, winner_token_id);
-        return ();
+    let (remove_relics_len) = loop_claim_order_relic(
+        0, realms_address, winner_realm_data, loser_token_id, loser_relics_len, 0, remove_relics
+    );
+
+    loop_remove_relic(0, 0, loser_relics_len, old_owner_token_id, remove_relics_len, remove_relics);
+    owned_relics_len.write(old_owner_token_id, loser_relics_len - remove_relics_len);
+    return ();
+}
+
+// @notice loop loser relic and return if order of winner matches holder
+// @implicit syscall_ptr
+// @implicit range_check_ptr
+// @param index: index of owner relics to check, starting at 0
+// @param realms_address: realms token address
+// @param winner_realm_data: realm data of the winner
+// @param loser_token_id: realm id of loser
+// @param loser_relics_len: length of relics array of loser
+// @param remove_relics_len: length of relics to remove from loser
+// @param remove_relics: array of relics to remove from loser
+func loop_claim_order_relic{pedersen_ptr: HashBuiltin*, syscall_ptr: felt*, range_check_ptr}(
+    index: felt,
+    realms_address: felt,
+    winner_realm_data: RealmData,
+    loser_token_id: Uint256,
+    loser_relics_len: felt,
+    remove_relics_len: felt,
+    remove_relics: Uint256*,
+) -> (remove_relics_len: felt) {
```

Comment:
Please rename the return value. It has the same name as a passed in argument which is weird.